### PR TITLE
chore(cli): Implement check for pod envoy proxy label

### PR DESCRIPTION
**Pull request by johnsonshi**
chore(cli): Implement check for pod envoy proxy label.

Addresses comments raised in #3881.

Signed-off-by: Johnson Shi <Johnson.Shi@microsoft.com>

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [x] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? No
    -   Did you notify the maintainers and provide attribution? NA

1. Is this a breaking change? No

### CODE REVIEW

File: pkg/cli/proxy_get_test.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func TestIsMeshedPod(t *testing.T) {
 			pod: corev1.Pod{
 				ObjectMeta: metav1.ObjectMeta{
 					Name:   "pod-1",
-					Labels: map[string]string{constants.EnvoyUniqueIDLabelName: "test"},
+					Labels: map[string]string{constants.EnvoyUniqueIDLabelName: "01066be8-c96b-4136-9e93-4c852d4137ea"},
```
Comment (johnsonshi) [inline pkg/cli/proxy_get_test.go:25]: Addresses https://github.com/openservicemesh/osm/pull/3881#discussion_r679558826

File: pkg/cli/proxy_get.go [hunk 2]
```diff
@@ -65,9 +65,11 @@ func GetEnvoyProxyConfig(clientSet kubernetes.Interface, config *rest.Config, na
 	return envoyProxyConfig, nil
 }
 
-// proxyLabelExists returns a boolean indicating if the pod is part of a mesh
+// proxyLabelExists returns a boolean indicating if the pod is part of a mesh.
+// Note that this is NOT a definitive check indicating pod membership in a mesh
+// as the label may continue to exist even if the pod is removed from the mesh.
 func proxyLabelExists(pod corev1.Pod) bool {
-	// osm-controller adds a unique label to each pod that belongs to a mesh
+	// osm-controller adds a unique label to a pod when it is added to a mesh
```
Comment (johnsonshi) [inline pkg/cli/proxy_get.go:72]: Addresses @SanyaKochhar 's comments in https://github.com/openservicemesh/osm/pull/3881#discussion_r680086117 and https://github.com/openservicemesh/osm/pull/3881#discussion_r680089278

File: pkg/cli/proxy_get_test.go (hunk 1 above)
Comment (shashankram) [reply pkg/cli/proxy_get_test.go:25]: Use `uuid.New().String()` here instead of hardcoding it.

Comment (shashankram) [reply pkg/cli/proxy_get_test.go:25]: Already being fixed correctly in https://github.com/openservicemesh/osm/pull/3900. We can drop this change and rebase onto https://github.com/openservicemesh/osm/pull/3900 once merged.

**Review by allenlsy — APPROVED** [review]

File: pkg/cli/proxy_get_test.go (hunk 1 above)
Comment (johnsonshi) [reply pkg/cli/proxy_get_test.go]: Changes dropped.

File: pkg/cli/proxy_get.go (hunk 2 above)
Comment (SanyaKochhar) [reply pkg/cli/proxy_get.go:72]: @johnsonshi could you expose this function for reuse in osm-health?

**Review by draychev — APPROVED** [review]

**Comment by johnsonshi** [comment]
This should now be exportable through `openservicemesh/osm/pkg/cli`

File: pkg/cli/proxy_get.go [hunk 3]
```diff
@@ -65,9 +65,11 @@ func GetEnvoyProxyConfig(clientSet kubernetes.Interface, config *rest.Config, na
...
-func proxyLabelExists(pod corev1.Pod) bool {
-	// osm-controller adds a unique label to each pod that belongs to a mesh
+// ProxyLabelExists returns a boolean indicating if the pod is part of a mesh.
+// Note that this is NOT a definitive check indicating pod membership in a mesh
+// as the label may continue to exist even if the pod is removed from the mesh.
+func ProxyLabelExists(pod corev1.Pod) bool {
+	// osm-controller adds a unique label to a pod when it is added to a mesh
 	proxyUUID, proxyLabelSet := pod.Labels[constants.EnvoyUniqueIDLabelName]
 	return proxyLabelSet && isValidUUID(proxyUUID)
 }
```
Comment (shashankram) [inline pkg/cli/proxy_get.go:75]: These sort of generic utility functions can live in `pkg/mesh`. It shouldn't have to be exported from the CLI pkg if it's meant to be reused outside the CLI. `pkg/cli` pertains to osm CLI utilities.

Comment (johnsonshi) [reply pkg/cli/proxy_get.go]: @shashankram Moved to `pkg/mesh`

**Pull request merged**